pkg/controller/mighook/predicate.go

```diff
@@ -0,0 +1,33 @@
+package mighook
+
+import (
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	"reflect"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type HookPredicate struct {
+	predicate.Funcs
+}
+
+func (r HookPredicate) Create(e event.CreateEvent) bool {
+	return true
+}
+
+func (r HookPredicate) Update(e event.UpdateEvent) bool {
+	old, cast := e.ObjectOld.(*migapi.MigHook)
+	if !cast {
+		return true
```

Comment:
If the old/new object in the `Event` cannot be cast to a `MigHook`, we don't want to queue the reconcile event.
```
return false
```
This is incorrect in many of the other _existing_ controller as well.